File: problems/0027-remove-element/solution.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut count = 0;

        nums.retain(|&n| {
            if (n == val) {
                return false;
            } else {
                count += 1;
                return true;
            }
        });

        return count;
    }
}

File: problems/0027-remove-element/solution.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        ans = 0
        for i in range(len(nums)-1, -1, -1):
            if nums[i] == val:
                ans += 1
                nums[i], nums[-ans] = nums[-ans], nums[i]
        
        return len(nums) - ans
